migrating/checklist/ossm-migrating-premigration-checklist-assembly.adoc

```diff
@@ -0,0 +1,212 @@
+<2> Network policy management has been disabled.
+<3> Addons and tracing have all been disabled.
+<4> Your `ServiceMeshControlPlane` is configured to use external metrics and tracing providers.
+<5> Managed gateways are disabled.
+<6> Either `MultiTenant` or `ClusterWide`.
+
+You will also have a `Telemetry` resource in your root namespace e.g. `istio-system` that looks roughly like this:
+
+[source,yaml]
+----
+apiVersion: telemetry.istio.io/v1
+kind: Telemetry
+metadata:
+  name: mesh-default
+  namespace: istio-system
+spec:
+  metrics: <1>
+    - providers:
+        - name: prometheus
+  tracing: <2>
+    - providers:
+        - name: otel
+----
+
+<1> Your metrics provider is specified. The `name` should match what is specified on your `ServiceMeshControlPlane` in the `spec.meshConfig.extensionProviders` field.
+<2> Your tracing provider is specified. The `name` should match what is specified on your `ServiceMeshControlPlane` in the `spec.meshConfig.extensionProviders` field.
+
+And a `Kiali` resource that looks roughly like this:
+
+[source,yaml]
+----
+apiVersion: kiali.io/v1alpha1
+kind: Kiali
+metadata:
+  name: kiali
+  namespace: istio-system
+spec:
+  version: default <1>
+  external_services:
+    prometheus: <2>
+      auth:
+        type: bearer
+        use_kiali_token: true
+      thanos_proxy:
+        enabled: true
+      url: https://thanos-querier.openshift-monitoring.svc.cluster.local:9091
+    tracing: <3>
+      enabled: true
+      provider: tempo
+      use_grpc: false
+      internal_url: http://tempo-sample-query-frontend.tempo:3200
+      external_url: https://tempo-sample-query-frontend-tempo.apps-crc.testing
+    grafana: <4>
+      enabled: false
+----
+
+<1> You are using the latest version of `Kiali`. As long as you install the 3.0 Service Mesh operator before upgrading Kiali, this version is compatible with both the 2.6 and 3.0 controlplanes.
+<2> Kiali is configured to use the external prometheus.
+<3> Kiali is configured to use the external tracing store.
+<4> Grafana configuration is disabled. Grafana is not supported with {SMProduct} 3.0.
+
+== Final check before migrating your deployment and workloads
+
+* [ ] Run the [migration-checker script](migration-checker.sh) to detect any issues with your environment.
+
+
+//01/23/2025: migration-checker script will be a specific GitHub link. Link to be determined just prior to release, so this checkbox will need to be updated accordingly. Check with Nick Fox for the URL to use. Nick is an engineer on Service Mesh.
+
+== Determine deployment model
+
+If the migration checker shows no errores, you are now you are ready to migrate.
+
+Your deployment model determines your migration path. Run the following command to determine if you are running a `MultiTenant` or a `ClusterWide` mesh:
+
+[source,terminal]
+----
+oc get smcp <smcp-name> -n <smcp-namespace> -o jsonpath='{.spec.mode}'
+----
+
```

Comment:
You can inspect your `ServiceMeshControlPlane` to determine if you are using cert-manager or not
```
...
spec:
  security:
    certificateAuthority:
      ...
      type: cert-manager
```

or use the following command
```
oc get smcp <smcp-name> -n <smcp-namespace> -o jsonpath='{.spec.security.certificateAuthority.type}'
```

If your `ServiceMeshControlPlane` is configured with cert-manager, in the output you will see:

```
cert-manager
```